packages/mux-player/src/utils.js

```diff
@@ -0,0 +1,162 @@
+/**
```

Comment:
yes I wanted to bring this up. is this something we are really set on?
is media-chrome also going to be Typescript?

Reply:
we have a lot of users who would like types, so I think we should aim for typescript unless there's a good reason not to. There have been asks for TypeScript support in media chrome, but given that @heff would also like to minimize build dependencies, one likely path forward would be either [JSDOC-style typing definitions](https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html) or external type definition files.

Reply:
I think it may be possible to generate a ts-style type def from jsdoc? Would make bridging the divide easier in that case.

Reply:
for me TS in jsdoc is really best of both worlds. 
yes, type definitions can be generated and to create a manual .d.ts file for most public API interfaces is not that hard.

the thing that annoys me with writing TS code is:

- adds noise to the code
- takes a bit longer to write and sometimes there are TS specific issues that take a chunk of your time
- runtime type errors still happen (https://blog.logrocket.com/is-typescript-worth-it/)
- reduction of bugs is less than it's hyped up to be (https://medium.com/javascript-scene/the-typescript-tax-132ff4cb175b)

I still like it but rather have it more as a linter and doc / autocompletion tool that gets out of the way.
Also a small argument could be it makes it harder for people to contribute.

Reply:
We decided on using TS for everything (and eventually circling back to port MC to TS). The reasoning is:

- Everyone wants types, writing in TS is the most reliable path to having good types
- The long term benefits we get from TS for maintainability, refactoring with confidence in the future, etc.

With the momentum TS has in the overall community of open source JS projects it feels like the right decision. That's just sort of a gut feeling.

I have definitely felt all of the "annoying points" that Wes brings up, but I still favor writing everything in TS. Even with those annoying points, I think the benefits of TS are still wroth it. Particularly when it comes to long term maintenance and eventually having to do significant refactors I find that TS gives me confidence.

Since MC started as JS I'm also not opposed to having MC use JS + JS-Doc/TS comments (also to @heff 's desire of keeping that project with minimal dependencies). My concern with that would be the possibility (probability?) of JS-Doc/TS comments becoming out of sync and causing the library to generate the wrong d.ts.

Reply:
Alright sounds good. Switching it up!